drivers/dma/dma_stm32f4x.c

```diff
@@ -275,7 +275,7 @@ static int dma_stm32_disable_stream(struct dma_stm32_device *ddata,
 				config &= ~DMA_STM32_SCR_EN);
 
 		/* After trying for 5 seconds, give up */
-		k_sleep(K_SECONDS(5));
+		k_sleep(5 * MSEC_PER_SEC);
```

Comment:
At one point in the past we had things like this `sec * MSEC_PER_SEC`, then `K_SECONDS()` was introduced and code was converted to use it, and now we are planning to go back. I would actually prefer the `K_SECONDS()` here as it is much more intuitive to use.

Reply:
I agree; the problem is it's being mis-used here.

`K_NO_WAIT`, `K_MSEC(x)`, `K_SECONDS(x)` are all currently documented to return a timeout value.  It happens that in the past all timeouts were in milliseconds, but that's no longer sufficient.  So the proposal on the table is to *some* but not all API that currently expects milliseconds (viz. `k_timer`, `k_poll`, but not `k_sleep`) to instead take a duration representation with better resolution.  This means API that produced timeouts will no longer produce a value that's suitable to `k_sleep()`.

There's no reason other than it's ugly that we can't have something like `K_MSEC_unit()` to indicate the translation is to milliseconds.

Or we could convert to C++ where all this would be much easier: `k_sleep(5_s)`

Regardless, whether `k_sleep()` switches to a timeout or remains milliseconds is a decision we'll have to consider.  I'll put this PR on hold until we get there.